core/node/auth/auth_impl.go

```diff
@@ -868,6 +1006,17 @@ func (ca *chainAuth) checkEntitlement(
 		return nil, err
 	}
 
+	// handle checking if the user is linked to a specific wallet
+	if args.kind == chainAuthKindIsWalletLinked {
+		for _, wallet := range wallets {
+			if wallet == args.walletAddress {
+				return boolCacheResult(true), nil
+			}
+		}
+		return boolCacheResult(false), nil
+	}
+
+	// populate the cache key for the linked wallets
```

Comment:
I don't love this because it mutates the args passed into the function. It concatenates the wallets into a string that is used as a cache key for several of the following operations. I guess at some point the string is split and the wallets are then used individually. I can take the comment out though.